sphinx/directives/other.py

```diff
@@ -39,6 +39,7 @@ class TocTree(Directive):
     final_argument_whitespace = False
     option_spec = {
         'maxdepth': int,
+        'name': str,
```

Comment:
I figured out the odd parsing stuff:

```
.. toctree::
    :maxdepth: 2

   getting_started
   versions
   builds
   features
   support
   faq
```

When you have a 4-indented option, and then a less indented list of index content, the option will get pulled up into the "title" with this new code.